src/algebraic_geometry/presheafed_space.lean

```diff
@@ -149,6 +158,15 @@ def forget : PresheafedSpace C ⥤ Top :=
 { obj := λ X, (X : Top.{v}),
   map := λ X Y f, f.base }
 
+/--
+The restriction of a presheafed space along an open embedding into the space.
+-/
+@[simps]
+def restrict {U : Top} (X : PresheafedSpace C)
+  (f : U ⟶ (X : Top.{v})) (h : open_embedding f) : PresheafedSpace C :=
+{ carrier := U,
+  presheaf := h.is_open_map.functor.op ⋙ X.presheaf }
```

Comment:
If we're moving to `has_limits` being a Prop, then surely this restriction construction should instead give you the existence of a presheaf satisfying a universal property and you should have to apply AC to make the data ;-)

Reply:
No thanks. :-)